src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +741,130 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPattern {
+    Ref(DatasetRef),
+    Pattern(Option<AccountName>, DatasetNamePattern),
+}
+
+impl DatasetRefPattern {
+    pub fn has_wildcard(s: &str, wildcard_symbol: char) -> bool {
+        s.contains(wildcard_symbol)
+    }
```

Comment:
It's used only in the `FromStr` implemetation. Perhaps, we can just call `contains` directly on the string.